src/framework/step/mod.rs

```diff
@@ -0,0 +1,132 @@
+use crate::framework::Error;
+#[cfg(feature = "enable-serde")]
+use serde::{
+    de::{Error as SerdeError, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use std::fmt::Formatter;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Step {
```

Comment:
I feel that I'd like to see a top level structure like this

```rust
pub enum Step {
    Test(TestStep),
     IPA(IPAStep)
}

pub enum IPAStep {
   ConvertShares(ModulusConversion),
   Sort(...),
   Attribute(...),
   ApplyDifferentialPrivacy(...)
}
```

thoughts?

Reply:
I don't think that we want that.  What we want is an implementation of a trait that can take an arbitrary input and produce a path (component) that we can use to route.  What @thurstonsand has is entirely fixed, which is a fine starting point, but we probably need to parameterize.

Then we would have test code with 

```rust
enum TestStep {
  Step1,
  Whatever { something more, or something, }
}
impl Step for TestStep { 
  fn to_path(&self) -> ...
```

And IPA, with its own structure, much like how Alex did it.

Those can coexist on the server, but on different base paths.

Reply:
i can create a trait with `to_path` and `from_path` and have all steps implement that